src/Libraries/CoreNodeModels/DefineData.cs

```diff
@@ -9,28 +8,62 @@
 using Newtonsoft.Json;
 using ProtoCore.AST.AssociativeAST;
 using VMDataBridge;
+using static DSCore.Data;
 
 
 namespace CoreNodeModels
 {
     [NodeName("DefineData")]
     [NodeDescription(nameof(Properties.Resources.RememberDescription), typeof(Properties.Resources))]
     [NodeCategory("Core.Data")]
-    [InPortNames(">")]
-    [InPortTypes("var[]..[]")]
-    [InPortDescriptions(typeof(Properties.Resources),
-        nameof(Properties.Resources.RememberInputToolTip))]
     [OutPortNames(">")]
     [OutPortTypes("var[]..[]")]
     [OutPortDescriptions(typeof(Properties.Resources), nameof(Properties.Resources.RememberOuputToolTip))]
     [IsDesignScriptCompatible]
     public class DefineData : DSDropDownBase
     {
-        private bool context;
         private List<DynamoDropDownItem> serializedItems;
+        private bool isAutoMode;
+        private bool isList;
 
+        /// <summary>
+        /// The IsAutoMode property enables the node to automatically validate and process input data.
+        /// AutoMode = true: the node checks input types for serialization compatibility, supports single values and non-nested lists,
+        /// and distinguishes between homogeneous and certain heterogeneous collections through inheritance.
+        /// Invalid or unsupported data types result in error messages,
+        /// while successful validation updates node properties and UI elements to reflect the processed data.
+        /// AutoMode = false: the node enters a manual processing mode,
+        /// where it strictly validates that the TypeID and Context predefined on the node match the attached input data.
+        /// Mismatches in expected data types or contexts—such as receiving a list instead of a single item,
+        /// or input data not matching the specified TypeID—result in errors, without automatically adjusting node settings.
+        /// This manual mode maintains the node's current configurations, ensuring an output is passed only when valid data is processed,
+        /// and retains the node's state in warning without resetting selections for invalid data.
+        /// </summary>
         [JsonProperty]
-        public bool IsList { get; set; }
+        public bool IsAutoMode
```

Comment:
@dnenov @twastvedt @saintentropy when this node's data types do not match and errors are thrown, does this node return null?

Does it make sense to explore - not returning anything and to stop all downstream computation? For example, this node could return an imperative AST like scopeIF.

Anyhow, that may be out of scope and we could do it later - it just seems like a large potential waste of compute if this node is used upstream of the entire graph in many cases.

Reply:
Yes, it does (should) return null/nothing. Makes sense to me. Is that (stopping downstream computation) something that other nodes do? If it's not complicated, perhaps we can look at it now. Things left for later tend not to get done.

Reply:
If you point me to an existing implementation, I can take that on, up to you @mjkkirschner and @twastvedt !

Reply:
@twastvedt no - returning null does not stop downstream computation - it passes null and likely leads to a bunch of slow exceptions for all downstream node calls - that's why I am suggesting rethinking that.

Reply:
look here - but it's not simple, if it's something vera team is interested in it probably warrants a meeting as it's a new scope/feature for sure.

https://github.com/DynamoDS/Dynamo/blob/5fe7dac7d79d18142887901d70b1ea8a4c6ba671/src/Libraries/CoreNodeModels/Logic/ScopedIf.cs#L15